src/VisualStudio/CSharp/Impl/Options/Formatting/FormattingOptionPage.cs

```diff
@@ -9,9 +9,18 @@ namespace Microsoft.VisualStudio.LanguageServices.CSharp.Options.Formatting
     [GuidAttribute(Guids.CSharpOptionPageFormattingGeneralIdString)]
     internal class FormattingOptionPage : AbstractOptionPage
     {
+        private FormattingOptionPageControl _optionPageControl;
         protected override AbstractOptionPageControl CreateOptionPage(IServiceProvider serviceProvider)
         {
-            return new FormattingOptionPageControl(serviceProvider);
+            _optionPageControl = new FormattingOptionPageControl(serviceProvider);
+            return _optionPageControl;
+        }
+
+        public override void SaveSettingsToStorage()
+        {
+            _optionPageControl.SetCodeCleanupAsConfigured();
+
+            base.SaveSettingsToStorage();
```

Comment:
who saves formatting options?

Reply:
The SaveSettingsToStorage() get called when click the Ok button on the option dialog.
https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.dialogpage.savesettingstostorage?view=visualstudiosdk-2017#Microsoft_VisualStudio_Shell_DialogPage_SaveSettingsToStorage